src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -179,7 +83,9 @@ typedef struct client_channel_channel_data {
   /** retry throttle data */
   grpc_server_retry_throttle_data* retry_throttle_data;
   /** maps method names to method_parameters structs */
-  grpc_slice_hash_table* method_params_table;
+  grpc_core::RefCountedPtr<grpc_core::SliceHashTable<
```

Comment:
This looks like a prime candidate for `auto`. Or, at a minimum, a `using MethodParamsTable = grpc_core::SliceHashTable<grpc_core::RefCountedPtr<ClientChannelMethodParams>>`, and thus `grpc_core::RefCountedPtr<MethodParamsTable> method_params_table`